pkg/secrets-store/nodeserver.go

```diff
@@ -180,20 +182,27 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		return nil, err
 	}
 
-	// mount before providers can write content to it
-	// In linux Mount tmpfs mounts tmpfs to targetPath
-	// In windows Mount tmpfs checks if the targetPath exists and if not, will create the target path
-	// https://github.com/kubernetes/utils/blob/master/mount/mount_windows.go#L68-L71
-	err = ns.mounter.Mount("tmpfs", targetPath, "tmpfs", []string{})
-	if err != nil {
-		errorReason = internalerrors.FailedToMount
-		klog.ErrorS(err, "failed to mount", "pod", klog.ObjectRef{Namespace: podNamespace, Name: podName})
-		return nil, err
-	}
-	mounted = true
 	var objectVersions map[string]string
-	if objectVersions, errorReason, err = ns.mountSecretsStoreObjectContent(ctx, providerName, string(parametersStr), string(secretStr), targetPath, string(permissionStr), podName); err != nil {
-		return nil, fmt.Errorf("failed to mount secrets store objects for pod %s/%s, err: %v", podNamespace, podName, err)
+	if republish {
+		if objectVersions, errorReason, err = ns.updateMountSecretsStoreObjectContent(ctx, providerName, string(parametersStr), string(secretStr), targetPath, string(permissionStr), podName); err != nil {
```

Comment:
We should be able to use `mountSecretsStoreObjectContent` as is because the calls to provider are idempotent.